HARK/core.py

```diff
@@ -327,6 +329,164 @@ def __str__(self):
+                except:
+                    raise ValueError("No constructor found for " + key)
+
+                # Get the names of arguments for this constructor and try to gather them
+                args_needed = get_arg_names(constructor)
+                has_no_default = {
+                    k: v.default is inspect.Parameter.empty
+                    for k, v in inspect.signature(constructor).parameters.items()
+                }
+                temp_dict = {}
+                any_missing = False
+                for j in range(len(args_needed)):
+                    this_arg = args_needed[j]
+                    if hasattr(self, this_arg):
+                        temp_dict[this_arg] = getattr(self, this_arg)
+                    else:
+                        try:
+                            temp_dict[this_arg] = self.parameters[this_arg]
+                        except:
+                            if has_no_default[this_arg]:
+                                # Record missing key-data pair
+                                any_missing = True
+                                missing_key_data.append((key, this_arg))
+
+                # If all of the required data was found, run the constructor and
+                # store the result in parameters (and on self)
+                if not any_missing:
+                    temp = constructor(**temp_dict)
+                    setattr(self, key, temp)
+                    self.parameters[key] = temp
+                    keys_complete[i] = True
+                    anything_accomplished_this_pass = True  # We did something!
+                else:
+                    pass  # Don't do anything, constructor will surely fail
+
+            # Check whether another pass should be performed
+            any_keys_incomplete = np.any(np.logical_not(keys_complete))
+            go = any_keys_incomplete and anything_accomplished_this_pass
+
+        # Store missing key-data pairs and exit
+        self._missing_key_data = missing_key_data
+        return
+
+    def describe_constructors(self, *args):
+        """
+        Prints to screen a string describing this instance's constructed objects,
+        including their names, the function that constructs them, the names of
+        those functions inputs, and whether those inputs are present.
+
+        Parameters
+        ----------
+        *args : str
+            Optional list of strings naming constructed inputs to be described.
+            If none are passed, all constructors are described.
+
+        Returns
+        -------
+        None.
+        """
+        if len(args) > 0:
+            keys = args
+        else:
+            keys = list(self.constructors.keys())
+        N_keys = len(keys)
+        yes = "\u2713"
+        no = "X"
+        maybe = "*"
+        noyes = [no, yes]
+
+        out = ""
+        for i in range(N_keys):
+            key = keys[i]
+            has_val = hasattr(self, key)
```

Comment:
could be looking up in parameters?

Reply:
Yes, `construct()` does this but I forgot to put that line in here. Will fix.